src/audio/host.c

```diff
@@ -462,6 +462,24 @@ static int host_trigger(struct comp_dev *dev, int cmd)
 						   hd->period_bytes);
 		}
 		break;
+	case COMP_TRIGGER_RELEASE:
+		if (dev->params.direction == SOF_IPC_STREAM_CAPTURE) {
+			struct dma_chan_status status;
+
+			buffer_reset_pos(hd->dma_buffer);
+			dma_status(hd->dma, hd->chan,
+				   &status, SOF_IPC_STREAM_CAPTURE);
+			if (status.r_pos !=
+			    ((uint32_t)hd->dma_buffer->r_ptr & 0x00FFFFFF)) {
+				/* hardcode here!!!
+				 * we didnot need to do consume operation,
+				 * just adjust the buffer pointer to sync
+				 * with host dma's register pointer.
+				 */
+				hd->dma_buffer->r_ptr += hd->dma_buffer->size;
```

Comment:
The comments say what you are doing but dont say why ?

Reply:
OK. I will update it in the commit description.
Let me describe it over here before:
the host buffer pointer should be sync with host dma pointer. because we did not stop the host dma, we have to read the DGBRP pointer first, then adjust the host buffer pointer according to the DGBRP pointer value.